### PR TITLE
Fixed #28358 -- Added LazyObject.__getattribute__() to mimic better the wrapped.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/28358

### CODE REVIEW

File: django/utils/functional.py [hunk 1]
```diff
@@ -234,7 +235,23 @@ def __init__(self):
         # override __copy__() and __deepcopy__() as well.
         self._wrapped = empty
 
-    __getattr__ = new_method_proxy(getattr)
+    def __getattribute__(self, name):
+        """
+        Return attribute from wrapped if
+        - attribute is not found on wrapper,
```
Comment (timgraham) [inline django/utils/functional.py]: Can you add some details about what situations each of these cases represent?

File: django/utils/functional.py [hunk 2]
```diff
@@ -234,7 +235,23 @@ def __init__(self):
...
+        - attribute is found on wrapper, but has special flag set.
+        """
+        __getattribute__ = super().__getattribute__
+        found = True
+        try:
+            val = __getattribute__(name)
+        except AttributeError:
+            found = False
+        if not found or not getattr(val, '_mask_wrapped', True):
+            if __getattribute__('_wrapped') is empty:
```
Comment (timgraham) [inline django/utils/functional.py]: Some inline comments here would be nice.

File: tests/utils_tests/test_lazyobject.py [hunk 3]
```diff
@@ -288,6 +288,26 @@ def test_deepcopy_class_no_evaluation(self):
         self.assertIs(obj._wrapped, empty)
         self.assertIs(obj2._wrapped, empty)
 
+    def test_getattribute(self):
+        attrs = [
```
Comment (timgraham) [inline tests/utils_tests/test_lazyobject.py]: Is an accurate docstring something like `"""Proxy methods don't exist on wrapped objects unless they're set."""`?

**Comment by timgraham** [comment]
Sergey, did you forget about this PR?

**Comment by felixxm** [comment]
Closing due to lack of follow up.

**Pull request closed without merging**